Drivers/Periphs/Src/SDCard.c

```diff
@@ -0,0 +1,104 @@
+
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h> //for va_list var arg functions
+#include "fatfs.h"
+#include "SDCard.h"
+#include "main.h"
+
+FRESULT fres;   //Result after operations
+
+UART_HandleTypeDef *huart;  //pointer to UART handler
+
+void SDCard_Init(UART_HandleTypeDef *uartBus){
+    huart = uartBus;
+}
+
+void myprintf(const char *fmt, ...){
+    static char buffer[256];
+    va_list args;
+    va_start(args, fmt);
+    vsnprintf(buffer, sizeof(buffer), fmt, args);
+    va_end(args);
+
+    int len = strlen(buffer);
+    HAL_UART_Transmit(huart, (uint8_t*)buffer, len, -1);
+}
+
+void SDCard_OpenFileSystem(FATFS FatFs){
+    //mount the drive
+    fres = f_mount(&FatFs, "", 1); //1=mount now
+    if (fres != FR_OK) {
+  	myprintf("f_mount error (%i)\r\n", fres);
+  	while(1);
+    }
+}
+
+void SDCard_GetStatistics(){
+    DWORD free_clusters;
+    DWORD free_sectors;
+    DWORD total_sectors;
+    FATFS *getFreeFs;
+
+    fres = f_getfree("", &free_clusters, &getFreeFs);
+    if(fres != FR_OK){
+        myprintf("f_getfree error (%i)\r\n", fres);
+        while(1);
+    }
+
+    //Formula comes from ChaN's documentation
```

Comment:
How do we find ChaN's documentation in case we need to reference it in the future? Can we just google "ChaN documentation"?

Reply:
A link to ChaN's documentation can be found at the top of the user_diskio_spi.c file.